analyzer/ldap/ldap.spicy

```diff
@@ -127,9 +127,58 @@ public type Result = unit {
 
 #-----------------------------------------------------------------------------
 public type Messages = unit {
-  : Message[];
+  : MessageWrapper[];
 };
 
+#-----------------------------------------------------------------------------
+public type MessageWrapper = unit {
+  # A wrapper around 'Message'. First, we try to parse a Message unit.
+  # There are two possible outcomes:
+  #   (1) Success -> We consumed all bytes and successfully parsed a Message unit
+  #   (2) No success -> self.backtrack() is called in the Message unit,
+  #                     so effectively we didn't consume any bytes yet.
+  # The outcome can be determined by checking the `success` variable of the Message unit
+
+  # This success variable is different, because this keeps track of the status for the MessageWrapper object
+  var success: bool = False;
+
+  # Here, we try to parse the message...
+  : Message &try {
+
+      # ... and only if the Message unit successfully parsed, we can set
+      # the status of this MessageWrapper's success to 'True'
+      if ( $$.success == True ) {
+        self.success = True;
+      }
+  }
+
+  # If we failed to parse the message, then we're going to scan the remaining bytes for the '\x30'
+  # start byte and try to parse a Message starting from that byte. This effectively
+  # strips the SASL layer if SASL Signing was enabled. Until now, I haven't found A
+  # better way to scan / determine the exact SASL header length yet, so we'll stick with this
+  # for the time being. If the entire LDAP packet was encrypted with SASL, then this will still fail and raise
+  # an error, which is expected (cause we can't decrypt encrypted SASL).
+
+  : bytes &until=b"\x30" if ( self.success == False);
+  var remainder: bytes;
+
+  # For some reason, the &until already consumes the '\x30', while it shouldn't do that.
+  # I think that's a bug in Spicy. For now, re-add it to the remainder.
+
+  : bytes &eod if ( self.success == False )  {
+    self.remainder = b"\x30" + $$;
+  }
+
+  # Again, try to parse a Message unit. Be aware that in theory (and probably in practice) this will
+  # sometimes fail if the '\x30' byte is also present in the SASL header.
+
+  # Also, we could try to do this recursively or try a few iterations, but for now I would suggest
+  # to try this extra parsing once to get the best cost/benefit tradeoff.
+  : Message &parse-from=self.remainder if ( self.success == False );
+
+};
```

Comment:
We could convert this to a `Message` when we are done so we do not expose the wrapping implementation detail outside of its concrete uses,
```suggestion
  message : Message &parse-from=self.remainder if ( self.success == False );
} &convert=self.message;
```

Reply:
Yes, I changed this